app/src/main/java/free/rm/skytube/gui/fragments/SubscriptionsFeedFragment.java

```diff
@@ -193,38 +194,30 @@ public static boolean isFlagSet(String flag) {
 	}
 
 	private void showNotification() {
-		// Sets an ID for the notification, so it can be updated.
-		if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
-			NotificationManager mNotificationManager =
-					(NotificationManager) getContext().getSystemService(Context.NOTIFICATION_SERVICE);
-
-			NotificationChannel mChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID, NOTIFICATION_CHANNEL_NAME, NotificationManager.IMPORTANCE_HIGH);
-			mChannel.setSound(null,null);
-			// Create a notification and set the notification channel.
-			Notification notification = new Notification.Builder(getContext(), NOTIFICATION_CHANNEL_NAME)
-					.setSmallIcon(R.drawable.ic_notification_icon)
-					.setContentTitle(getString(R.string.fetching_subscription_videos))
-					.setContentText(String.format(getContext().getString(R.string.fetched_videos_from_channels), numVideosFetched, numChannelsFetched, numChannelsSubscribed))
-					.setChannelId(NOTIFICATION_CHANNEL_ID)
-					.build();
-			mNotificationManager.createNotificationChannel(mChannel);
-
-// Issue the notification.
-			mNotificationManager.notify(NOTIFICATION_ID , notification);
-		} else {
-			final Intent emptyIntent = new Intent();
-			PendingIntent pendingIntent = PendingIntent.getActivity(getContext(), 1, emptyIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-			NotificationCompat.Builder builder = new NotificationCompat.Builder(getContext(), "")
-					.setSmallIcon(R.drawable.ic_notification_icon)
-					.setContentTitle(getString(R.string.fetching_subscription_videos))
-					.setContentText(String.format(getContext().getString(R.string.fetched_videos_from_channels), numVideosFetched, numChannelsFetched, numChannelsSubscribed))
-					.setPriority(NotificationCompat.FLAG_ONGOING_EVENT)
+		final NotificationManager notificationManager = ContextCompat.getSystemService(requireContext(),
+				NotificationManager.class);
+		final NotificationCompat.Builder builder = new NotificationCompat.Builder(requireContext(), NOTIFICATION_CHANNEL_ID)
+				.setSmallIcon(R.drawable.ic_notification_icon)
+				.setContentTitle(getString(R.string.fetching_subscription_videos))
+				.setContentText(String.format(getString(R.string.fetched_videos_from_channels),
+						numVideosFetched, numChannelsFetched, numChannelsSubscribed));
+
+		if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+			PendingIntent pendingIntent = PendingIntent.getActivity(requireContext(),
+					1, new Intent(), PendingIntent.FLAG_UPDATE_CURRENT);
+			builder.setPriority(NotificationCompat.FLAG_ONGOING_EVENT)
 					.setContentIntent(pendingIntent);
+		}
 
-
-			NotificationManager notificationManager = (NotificationManager) getContext().getSystemService(Context.NOTIFICATION_SERVICE);
-			notificationManager.notify(NOTIFICATION_ID, builder.build());
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
Any particular reason to use a separate if, instead of a plain simple 'else' ?

Reply:
My bad, I'll change it.